lsp/nls/src/linearization/mod.rs

```diff
@@ -460,14 +460,31 @@ impl<'a> Linearizer for AnalysisHost<'a> {
                 let Some(linearization) = lin.lin_cache.get(file) else {
                     return
                 };
-                // This is safe because the import file is resolved, before we linearize
-                // the containing file, therefore the cache MUST have the term stored.
+
+                // If the import is an external format (such as JSON or YAML), no position will be
+                // set. We could in fact set at least the position of the root term, but since this
+                // isn't the case currently, an unwrap() below would fail. For now, we simply bail
+                // out. This just means we can't "go to definition" on an external import for the
+                // time being.
+                if lin
+                    .cache
+                    .input_format(file)
+                    .map(|fmt| !matches!(fmt, InputFormat::Nickel))
+                    .unwrap_or(false)
+                {
+                    return;
+                }
+
+                // This is safe because the import file is resolved before we linearize the
+                // containing file, therefore the cache MUST have the term stored.
                 let term = lin.cache.get(*file).unwrap();
                 let position = final_term_pos(&term);
 
-                // This unwrap fails only when position is a `TermPos::None`, which only happens
-                // if the `RichTerm`, has been transformed or evaluated. None of these happen before
-                // linearization, so this is safe.
+                // unwrap(): this unwrap fails only when position is a `TermPos::None`, which only
+                // happens if the `RichTerm` has been transformed or evaluated. None of these
+                // happen before linearization. At this point, we also ensured that we are only
+                // considering a Nickel source (and not a term deserialized from e.g. JSON), and
+                // terms parsed from Nickel must have a position set. So
```

Comment:
Not that it really matters, but in essence we're `get`ting the term out of the linearization cache twice. First to check that the input format is `Nickel`. Then we throw away the entry and find it again, this time to just unwrap it. Maybe a `let else` construct could be better here?

Reply:
I think `cache.get` returns a `RichTerm`, without the metadata. We probably don't want to change this API, which is used in several places. What I might do is to either add a method that returns the whole `CachedTerm`, but I'm not sure it's a useful interface. Or one that returns a term together with the `input_format`. On the other hand, it's just one additional hashmap get by import, which is probably totally negligible.

Reply:
I agree, it's probably negligible. Let's not hold up this PR with that 👍

Reply:
In the end I added a method `get_with_input_format`, which looks reasonable.